src/GraphQL/Execution/ExecutionOptions.cs

```diff
@@ -12,22 +12,39 @@
 
 namespace GraphQL
 {
+    /// <summary>Configuration options to be passed to <see cref="IDocumentExecuter"/> to execute a query</summary>
     public class ExecutionOptions : IProvideUserContext
     {
+        /// <summary>Schema of graph to use; required<br/><br/>Schema will be initialized if has not yet been initialized.</summary>
         public ISchema Schema { get; set; }
+
+        /// <summary>Object to pass to the <see cref="IResolveFieldContext.Source"/> property of first-level resolvers</summary>
         public object Root { get; set; }
+
+        /// <summary>GraphQL query to parse and execute; required</summary>
         public string Query { get; set; }
+
+        /// <summary>GraphQL query operation name; defaults to first operation defined in query</summary>
         public string OperationName { get; set; }
+
+        /// <summary>Parsed GraphQL request; will parse <see cref="Query"/> if not set</summary>
```

Comment:
Here it is worth saying why both objects Document and Query are presented in options

Reply:
Good question ... why are they?  I noticed that if Query is null it will throw an exception, regardless of whether Document is set.  Not sure exactly how I should change this comment.

Reply:
I think the ability to set `Document` in options is to optimize performance. If there is some caching on the caller side, then it can set the already created and cached document so there will be no need in `_documentBuilder.Build(options.Query);`. But these are my assumptions. @joemcbride It is right?

Reply:
It is certainly a valid example.  I can add that into the comment.

Reply:
OK

Reply:
(Although in that example, they should probably provide a custom IDocumentBuilder to perform the caching.)

Reply:
Caching can be implemented anywhere, including `IDocumentBuilder`.

Reply:
How does the updated comment look now?

Reply:
I posted new suggestion.